src/Stripe.Tests.XUnit/BaseStripeTest.cs

```diff
@@ -0,0 +1,161 @@
+        {
+            return GetFixture(path, null);
+        }
+
+        protected static string GetFixture(string path, string[] expansions)
+        {
+            string url = $"http://localhost:{Port}{path}";
+
+            if (expansions != null)
+            {
+                string query = string.Join("&", expansions.Select(x => $"expand[]={x}").ToArray());
+                url += $"?{query}";
+            }
+
+            using (HttpClient client = new HttpClient())
+            {
+                client.DefaultRequestHeaders.Authorization
+                    = new System.Net.Http.Headers.AuthenticationHeaderValue(
+                        "Bearer",
+                        "sk_test_123"
+                    );
+
+                HttpResponseMessage response;
+
+                try
+                {
+                    response = client.GetAsync(url).Result;
+                }
+                catch (Exception)
+                {
+                    throw new Exception(
+                        $"Couldn't reach stripe-mock at `localhost:{Port}`. "
+                        + "Is it running? Please see README for setup instructions."
+                    );
+                }
+
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    throw new Exception(
+                        $"stripe-mock returned status code: {response.StatusCode}."
+                    );
+                }
+
+                return response.Content.ReadAsStringAsync().Result;
+            }
+        }
+
+        /// <summary>
+        /// Checks that stripe-mock is running and that it's a recent enough version.
+        /// </summary>
+        private static void CheckStripeMock()
+        {
+            Port = Environment.GetEnvironmentVariable("STRIPE_MOCK_PORT") ?? "12111";
+            string url = $"http://localhost:{Port}";
+
+            using (HttpClient client = new HttpClient())
+            {
+                HttpResponseMessage response;
+
+                try
+                {
+                    response = client.GetAsync(url).Result;
+                }
+                catch (Exception)
+                {
+                    throw new Exception(
+                        $"Couldn't reach stripe-mock at `localhost:{Port}`. "
+                        + "Is it running? Please see README for setup instructions."
+                    );
+                }
+
+                string version = response.Headers.GetValues("Stripe-Mock-Version").FirstOrDefault();
+
+                if (!(version.Equals("master")) &&
+                    (CompareVersions(version, MockMinimumVersion) > 0))
+                {
+                    throw new Exception(
+                        $"Your version of stripe-mock ({version}) is too old. The minimum "
+                        + $"version to run this test suite is {MockMinimumVersion}. Please see its "
+                        + "repository for upgrade instructions."
```

Comment:
OOC why don't we require the exact same version? New versions will likely break tests too (that happened to be at least twice in Java)

Reply:
Yes, but that's what you want:
1. Upgrade stripe-mock
2. See which tests fail, and fix the code (or the tests themselves)

If the version check prevents the test suite from running at all, then 2. becomes more difficult :)